File: its-my-training-day/leetcode-top-75/questions/bit-manipulation/bitwise/number-of-1-bits/solution.py
import os
import sys
from typing import List, Dict, Optional, Tuple

ROOT_DIR = os.path.abspath(os.path.join(os.path.dirname(__file__), "../../../../"))
if ROOT_DIR not in sys.path:
    sys.path.append(ROOT_DIR)

from shared.python.ds import ArrayList, ListNode, Stack, Queue, TreeNode, MinHeap, MaxHeap

def number_of_1_bits(n: int) -> int:
    count = 0
    while n != 0:
        n &= n - 1
        count += 1
    return count

File: its-my-training-day/leetcode-top-75/questions/bit-manipulation/bitwise/number-of-1-bits/solution.ts
import {
  ArrayList,
  Stack,
  Queue,
  ListNode,
  TreeNode,
  MinHeap,
  MaxHeap,
} from "../../../../shared/typescript/ds";

export const numberOf1Bits = (n: number): number => {
  let count = 0;
  let x = n >>> 0;
  while (x !== 0) {
    x &= x - 1;
    count += 1;
  }
  return count;
};
